sudoku.go

```diff
@@ -0,0 +1,103 @@
+package sudoku
+
+import "fmt"
+
+// Solve is the main task of this app
+func Solve(puzzle []int) ([]int, error) {
```

Comment:
from the test readme ```more suitable data structure``` could you give an example of an alternative data structure that could work.

Reply:
I think an easier data structure would have been `puzzle [9][9]int`. Loops simpler because we would treat it as a grid